cmake/AddChmTarget.cmake

```diff
@@ -1,36 +1,75 @@
-function (add_chm_target docname lang entities figures)
+#
+# A function to generate chm documentation
+#
+# FUNCTION:
+#   add_chm_target
+# ARGUMENTS:
+# - docname: The basename of the main xml file. Will be used to locate
+#            this primary xml file and for various output files/directories.
+#            Either "gnucash-guide" or "gnucash-help" now.
+# - lang: The language of the current document, such as "C", "de" and so on.
+# - entities: A list of all xml files this document is composed of. ONLY filename, WITHOUT path.
+#             It does NOT contain "${docname}.xml".
+# - figures: A list of FULL PATH image files.
+# - dtd_files: A list of FULL PATH DTD files.
 
-    set(chmfile "${docname}.chm")
-    set(mapfile "${docname}.hhmap")
+function (add_chm_target docname lang entities figures dtd_files)
 
-    set(BUILD_DIR "${DOCDIR_BUILD}/${lang}")
-    file(MAKE_DIRECTORY "${BUILD_DIR}")
 
-    file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/htmlhelp")
+    set(BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/htmlhelp")
+
+    # Copy figures for this document
+    file(MAKE_DIRECTORY "${BUILD_DIR}/figures")
+    add_custom_command(
+        OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/chmfigtrigger"
+        COMMAND ${CMAKE_COMMAND} -E copy ${figures} "${BUILD_DIR}/figures"
+        COMMAND touch "${CMAKE_CURRENT_BINARY_DIR}/figtrigger"
+        DEPENDS ${figures})
+
+    # Copy icon files for this documents
+    # Removed dependency because icon files are rarely changed. 
```

Comment:
> > I would not omit the dependency just because this rarely changes. It is still dependent on it and it may change. Not setting the dependency would in that case force users to reset their build system.
> 
> @gjanssens
> I rethought and I'll do it in another PR.

I made PR #215 to add a dependency to GnuCash icon files.